Makefile

```diff
@@ -1,34 +1,203 @@
+#
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# openshift.io/gitops-operator-bundle:$VERSION and openshift.io/gitops-operator-catalog:$VERSION.
+IMAGE_TAG_BASE ?= openshift.io/gitops-operator
 
-.PHONY: test-e2e
-test-e2e:
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
+all: build
+
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+##@ Development
+
+manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
+	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+vet: ## Run go vet against code.
+	go vet ./...
+
+test-all: manifests generate fmt vet ## Run all tests.
+	go test ./... -coverprofile cover.out
+
+test-e2e: manifests generate fmt vet ## Run e2e tests.
```

Comment:
```
Summarizing 1 Failure:

[Fail] GitOpsServiceController Validate non-default Argo CD instance [It] Create a non-default Argo CD instance 
/home/varshab/go/src/github.com/redhat-developer/gitops-operator/test/e2e/gitopsservice_test.go:198

Ran 51 of 51 Specs in 458.445 seconds
FAIL! -- 50 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestAPIs (458.45s)
FAIL
coverage: [no statements]
FAIL    github.com/redhat-developer/gitops-operator/test/e2e    458.476s
FAIL
make: *** [Makefile:92: test-e2e] Error 1
```
Tested locally

Reply:
It passes in CI - https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-developer_gitops-operator/136/pull-ci-redhat-developer-gitops-operator-master-v4.7-e2e/1425518253169446912/artifacts/e2e/e2e-steps/build-log.txt

Reply:
However i can reproduce the failure locally but not the failure that @varshab1210 reported above.

Reply:
@chetan-rns Can you please try locally once.

Reply:
Looks like a flake and will be tracked in https://issues.redhat.com/browse/GITOPS-1266